codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -45,10 +47,327 @@ pub enum Ecosystem {
     Fuel,
 }
 
+// Allows to get an env var with a lazy loading of .env file
+#[derive(Debug)]
+struct EnvState {
+    // Lazy loading of .env file
+    maybe_dotenv: Option<EnvMap>,
+    project_root: PathBuf,
+}
+
+impl EnvState {
+    fn new(project_root: &PathBuf) -> Self {
+        EnvState {
+            maybe_dotenv: None,
+            project_root: project_root.clone(),
+        }
+    }
+
+    fn var(&mut self, name: &str) -> Option<String> {
+        match std::env::var(name) {
+            Ok(val) => Some(val),
+            Err(_) => {
+                let result = match &self.maybe_dotenv {
+                    Some(env_map) => env_map.var(name),
+                    None => {
+                        match EnvLoader::with_path(self.project_root.join(".env"))
+                            .sequence(EnvSequence::InputOnly)
+                            .load()
+                        {
+                            Ok(env_map) => {
+                                self.maybe_dotenv = Some(env_map.clone());
+                                env_map.var(name)
+                            }
+                            Err(err) => {
+                                match err {
+                              dotenvy::Error::Io(_, _) => (),
+                              _ => println!("Warning: Failed loading .env file with unexpected error: {err}"),
+                          };
+                                self.maybe_dotenv = Some(EnvMap::new());
+                                Err(err)
+                            }
+                        }
+                    }
+                };
+                match result {
+                    Ok(val) => Some(val),
+                    Err(_) => None,
+                }
```

Comment:
So nothing is done with the error case? What about propagating the errors as they go?

Reply:
It's either not present or not a unicode. I've decided to change it to Option for simpler mocking in tests.

<img width="1257" alt="image" src="https://github.com/user-attachments/assets/7604950a-98b6-4751-8a61-3e2d3aff6263">